pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -366,10 +367,22 @@ func (cb *ClusterBuilder) buildDefaultCluster(name string, discoveryType cluster
 	ec := NewMutableCluster(c)
 	switch discoveryType {
 	case cluster.Cluster_STRICT_DNS, cluster.Cluster_LOGICAL_DNS:
-		if cb.supportsIPv4 {
+		if networkutil.AllIPv4(cb.proxyIPAddresses) {
+			// IPv4 only
 			c.DnsLookupFamily = cluster.Cluster_V4_ONLY
-		} else {
+		} else if networkutil.AllIPv6(cb.proxyIPAddresses) {
+			// IPv6 only
 			c.DnsLookupFamily = cluster.Cluster_V6_ONLY
+		} else {
+			// Dual Stack
+			if features.EnableDualStack {
+				// If dual-stack, it may be [IPv4, IPv6] or [IPv6, IPv4]
+				// using Cluster_ALL to enable Happy Eyeballsfor upstream connections
+				c.DnsLookupFamily = cluster.Cluster_ALL
```

Comment:
Curious if we want ALL or AUTO. Looks like curl and Golang both default to ALL equivilent so maybe it is a good default. Any opinions @kyessenov @ramaraochavali @jacob-delgado ?

Reply:
I think ALL seems to be a good default.